avalon/pipeline.py

```diff
@@ -1003,6 +995,7 @@ def _format_work_template(template, session=None):
         "root": registered_root(),
         "project": session["AVALON_PROJECT"],
         "silo": session["AVALON_SILO"],
+        "hierarchy": session['AVALON_HIERARCHY'],
```

Comment:
Will this error out on pipelines who haven't set `AVALON_HIERARCHY` yet?

1. Should this be in the Session schema if it's required?
2. What is this value and when would you set it? How would you compute it for any asset?

Reply:
yes I'd say that making is required in the schema is reasonable. however there is a caveat here. Because it hold multiple key in the path joined together with a path.sep, it will be quite hard to parse the opposite way. So for example what you are using in your colorbleed config (get context from path) will likely not work. 

AVALON_HIERARCHY resolve to a list of parents joined by path.sep so for example If asset is in this hierarchy: `project1/assets/characters/hero`, then the template would be `{project/{hierarchy}/{asset}. 

`hierarchy` = 'assets/characters`

Reply:
This needs to be made optional, as per the comments.